custom_components/entsoe/api_client.py

```diff
@@ -82,10 +82,16 @@ def query_day_ahead_prices(
                         if resolution != "PT60M":
                             continue
 
-                        start_time = period.find(".//timeInterval/start").text
+                        response_start = period.find(".//timeInterval/start").text
+                        start_time = (
+                            datetime.strptime(response_start, "%Y-%m-%dT%H:%MZ")
+                            .replace(tzinfo=pytz.UTC)
+                            .astimezone()
+                        )
 
-                        date = (
-                            datetime.strptime(start_time, "%Y-%m-%dT%H:%MZ")
+                        response_end = period.find(".//timeInterval/start").text
```

Comment:
This looks like a copy paste error. You are fetching the 'start' tag for the reponse_end time.

Reply:
Resolved,
I also found that the endtime should be excluded [from - end> as this value is given in a follow up period

![image](https://github.com/user-attachments/assets/c31f0d3d-d6ea-4986-968e-202d43000989)